module/motion/QuinticWalk/src/QuinticWalk.cpp

```diff
@@ -32,96 +34,132 @@ namespace module::motion {
+        config.params.foot_z_pause                  = cfg["walk"]["foot"]["z_pause"].as<float>();
+        config.params.foot_put_down_z_offset        = cfg["walk"]["foot"]["put_down"]["z_offset"].as<float>();
+        config.params.foot_put_down_phase           = cfg["walk"]["foot"]["put_down"]["phase"].as<float>();
+        config.params.foot_put_down_roll_offset     = cfg["walk"]["foot"]["put_down"]["roll_offset"].as<float>();
+        config.params.foot_apex_phase               = cfg["walk"]["foot"]["apex_phase"].as<float>();
+        config.params.foot_overshoot_ratio          = cfg["walk"]["foot"]["overshoot"]["ratio"].as<float>();
+        config.params.foot_overshoot_phase          = cfg["walk"]["foot"]["overshoot"]["phase"].as<float>();
+        config.params.trunk_height                  = cfg["walk"]["trunk"]["height"].as<float>();
+        config.params.trunk_pitch                   = 1.0f + cfg["walk"]["trunk"]["pitch"].as<Expression>();
+        config.params.trunk_phase                   = cfg["walk"]["trunk"]["phase"].as<float>();
+        config.params.trunk_x_offset                = cfg["walk"]["trunk"]["x_offset"].as<float>();
+        config.params.trunk_y_offset                = cfg["walk"]["trunk"]["y_offset"].as<float>();
+        config.params.trunk_swing                   = cfg["walk"]["trunk"]["swing"].as<float>();
+        config.params.trunk_pause                   = cfg["walk"]["trunk"]["pause"].as<float>();
+        config.params.trunk_x_offset_p_coef_forward = cfg["walk"]["trunk"]["x_offset_p_coef"]["forward"].as<float>();
+        config.params.trunk_x_offset_p_coef_turn    = cfg["walk"]["trunk"]["x_offset_p_coef"]["turn"].as<float>();
+        config.params.trunk_pitch_p_coef_forward =
+            1.0f + cfg["walk"]["trunk"]["pitch_p_coef"]["forward"].as<Expression>();
+        config.params.trunk_pitch_p_coef_turn = 1.0f + cfg["walk"]["trunk"]["pitch_p_coef"]["turn"].as<Expression>();
+        config.params.kick_length             = cfg["walk"]["kick"]["length"].as<float>();
+        config.params.kick_phase              = cfg["walk"]["kick"]["phase"].as<float>();
+        config.params.kick_vel                = cfg["walk"]["kick"]["vel"].as<float>();
+        config.params.pause_duration          = cfg["walk"]["pause"]["duration"].as<float>();
+
+
+        config.max_step.x() = cfg["max_step"]["x"].as<float>();
+        config.max_step.y() = cfg["max_step"]["y"].as<float>();
+        config.max_step.z() = cfg["max_step"]["z"].as<float>();
+        config.max_step_xy = cfg["max_step"]["xy"].as<float>();
+
+        config.imu_active          = cfg["imu"]["active"].as<bool>();
+        config.imu_pitch_threshold = 1.0f + cfg["imu"]["pitch"]["threshold"].as<float>();
+        config.imu_roll_threshold  = cfg["imu"]["roll"]["threshold"].as<float>();
+
+        for (int i = 0; i < ServoID::NUMBER_OF_SERVOS; ++i) {
+            if ((i >= 6) && (i < 18)) {
```

Comment:
Numbers are magic. Probably name these constants. idk.

Reply:
Changed magic numbers to actual servos, changed variable name

Reply:
Great work! I'll make one more suggestion to put the cream on top